File: ios/FileConfigReader.swift
enum FileConfigKeys : String {
     case RootDir = "rootDir", NestedDir = "nestedDir", FileName = "fileName"
}

class FileConfigReader {
    private let rootDir: String
    private let nestedDir: String?
    private let fileName: String

    init(rawFileConfig: [String: String]) throws {
        if (
            rawFileConfig[FileConfigKeys.RootDir.rawValue] == nil ||
            rawFileConfig[FileConfigKeys.FileName.rawValue] == nil
        ) {
            throw RFLError.fileConfiguration(FileConfigurationError.incomplete)
        }
        rootDir = rawFileConfig[FileConfigKeys.RootDir.rawValue]!
        nestedDir = rawFileConfig[FileConfigKeys.NestedDir.rawValue]
        fileName = rawFileConfig[FileConfigKeys.FileName.rawValue]!
    }

    func read() -> (rootDir: String, nestedDir: String?, fileName: String) {
        return (rootDir: rootDir, nestedDir: nestedDir, fileName: fileName)
    }
}

File: android/src/main/java/com/reduxfilelogger/utils/FileConfigReader.kt
package com.reduxfilelogger.utils

import com.facebook.react.bridge.ReadableMap
import com.reduxfilelogger.IncompleteFileConfigurationError

data class FileConfig(val rootDir: String, val nestedDir: String?, val fileName: String)

class FileConfigReader(rawFileConfig: ReadableMap) {
  private val fileConfig: FileConfig

  init {
    val rootDir = rawFileConfig.getString(FileConfigKeys.ROOT_DIR.key)
    val nestedDir = rawFileConfig.getString(FileConfigKeys.NESTED_DIR.key)
    val fileName = rawFileConfig.getString(FileConfigKeys.FILE_NAME.key)

    if (rootDir == null || fileName == null) {
      throw IncompleteFileConfigurationError()
    }

    this.fileConfig = FileConfig(rootDir, nestedDir, fileName)
  }

  fun read(): FileConfig {
    return this.fileConfig
  }

  companion object {
    enum class FileConfigKeys(val key: String) {
      ROOT_DIR("rootDir"),
      NESTED_DIR("nestedDir"),
      FILE_NAME("fileName")
    }
  }
}
